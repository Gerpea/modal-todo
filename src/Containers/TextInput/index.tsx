import React from 'react';
import clsx from 'clsx';

import style from './styles.module.scss';

export interface Props {
  className?: string;
  onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
  value?: string;
  placeholder?: string;
  error?: string;
}

function TextInput({ error, value, placeholder, className, onChange }: Props): React.ReactElement {
  const styles = clsx(className, style.Input, error && style.Error);

  return <input value={value} className={styles} onChange={onChange} placeholder={placeholder} />;
}

export default TextInput;
