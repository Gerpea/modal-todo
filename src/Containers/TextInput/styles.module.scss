@import '../../variables';

.Input {
  position: relative;

  outline: none;
  border: none;
  background-color: transparent;

  width: 100%;
  padding-top: 13px;
  padding-bottom: 11px;

  border-bottom: 1px solid transparentize($c-blue, 0.5);
  color: $c-black;

  font-size: 0.9rem;
  line-height: 1.15;
}

.Input::placeholder {
  color: transparentize($c-blue, 0.5);
}

.Input:focus {
  border-color: $c-blue;
}

.Error {
  border-color: $c-blue;
}
