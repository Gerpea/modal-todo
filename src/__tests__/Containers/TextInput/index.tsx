import React from 'react';
import renderer from 'react-test-renderer';

import TextInput from 'Containers/TextInput';

describe('Input test', () => {
  it('Input renders', () => {
    const tree = renderer.create(<TextInput onChange={() => void {}} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Input with error', () => {
    const tree = renderer.create(<TextInput onChange={() => void {}} error="error" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
