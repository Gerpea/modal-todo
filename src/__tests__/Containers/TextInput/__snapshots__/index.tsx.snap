// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input test Input renders 1`] = `
<input
  className="Input"
  onChange={[Function]}
/>
`;

exports[`Input test Input with error 1`] = `
<input
  className="Input Error"
  onChange={[Function]}
/>
`;
